packages/http-client/src/remote-index-api.ts

```diff
@@ -1,7 +1,29 @@
 import type { BaseQuery, IndexApi, Page, Pagination, StreamState } from '@ceramicnetwork/common'
+import { StreamUtils, fetchJson } from '@ceramicnetwork/common'
 
+/**
+ * IndexAPI implementation on top of HTTP endpoint.
+ */
 export class RemoteIndexApi implements IndexApi {
-  queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
-    throw new Error(`NotImplemented: RemoteIndexApi::queryIndex`)
+  private readonly _collectionURL: URL
+
+  constructor(apiUrl: URL, private readonly fetchJsonFn: typeof fetchJson = fetchJson) {
```

Comment:
why have `fetchJsonFn` as an argument?  It's easy enough to mock out the function in unit tests without this, and seems to just complicate the API.

Reply:
It _was_ easy. Jest has very minimal support for mocking of ES modules. The tradeoff is this non-fancy argument, or quite a few `await import` clauses, along with mocking fetch output per test. I tried that. Mocking fetch here is ugly.

Reply:
Okay, what about if we take it out of the constructor, but still have it as a private member variable that just always gets assigned to fetchJson at construction?  Then the unit tests can still override the function member, but we haven't expanded the public API

Reply:
It's a less obvious mess, but I am ok with it.

Reply:
I think this is the only change now I'm waiting on to re-review and approve this PR

Reply:
@stbrody Changed to private _fetchJson. Please, have a look!